### PR TITLE
Fixes #352 Add warning for incorrect using injectGlobal

### LINKED ISSUE #352: Consider adding a warnings for incorrect tagged template literal usage in development
I was adding styled-components to an app and global styles weren't showing up. There weren't any errors and it wasn't immediately obvious what I was doing wrong. It turned out I was using `injectGlobal` wrong:

```js
// wrong: calling injectGlobal as a function. oops.
injectGlobal(`html { color: red; }`)

// correct: using a tagged template literal
injectGlobal`html { color: red; }`
```

The first argument of tagged template literals includes a [`raw` property](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Raw_strings). Adding a check in development to prints out an error when `strings.raw` is missing would help catch this kind of mistake. I'm not sure how frequently this happens, but considering the relative rarity of tagged template literal usage, I figure it wouldn't hurt.

```js
if (process.env.NODE_ENV !== 'production' && !('raw' in strings)) {
  console.error('...')
}
```

### PR DESCRIPTION
**Pull request by maksugr**
Add warning for incorrect using injectGlobal. More: https://github.com/styled-components/styled-components/issues/352 
Ready for suggestions.

### CODE REVIEW

**Comment by maksugr** [comment]
Excuse me, some problems with this PR? Nobody watched it:)

**Comment by TrySound** [comment]
Just spent some time to debug why this doesn't work
```js
injectGlobal(`
  body {
    font-family: sans-serif;
    margin: 0;
  }
`);
```
This kind of errors will improve DX

File: src/constructors/injectGlobal.js [hunk 1]
```diff
@@ -5,6 +5,10 @@ import type { Interpolation, Stringifier } from '../types'
 
 export default (stringifyRules: Stringifier, css: Function) => {
   const injectGlobal = (strings: Array<string>, ...interpolations: Array<Interpolation>) => {
+    if (process.env.NODE_ENV !== 'production' && !(Array.isArray(strings))) {
+      // eslint-disable-next-line no-console
+      console.warn('Incorrect using injectGlobal as a function. Instead use it as a tagged template literal, see: https://www.styled-components.com/docs/api')
```
Comment (TrySound) [inline src/constructors/injectGlobal.js]: I think using `console.error` is more correct, since it's not deprecation or something like deoptimization.

**Comment by TrySound** [comment]
@maksugr You can test this case with temporary patching `console.error` before using injectGlobal. Don't forget to reverse

**Comment by probablyup** [comment]
I’d rather see this land in an eslint plugin.

**Comment by kitten** [comment]
@probablyup I agree with you.

@maksugr sorry, this is mostly the reason why I haven’t pushed for this PR or a couple of past ones.

It’s not realistic to add warnings for every single misuse unfortunately. We have already made an effort to add a lot of new warnings that are very explicit but almost none of that refer to misuse of the API :/

Thanks for your effort either way :+1:

**Pull request closed without merging**